src/uct/sm/mm/mm_posix.c

```diff
@@ -276,23 +438,41 @@ static ucs_status_t uct_posix_free(void *address, uct_mm_id_t mm_id, size_t leng
         goto err;
     }
 
-    /* use the mmid (63 bits uuid) to recreate the file_name for shm_unlink */
-    sprintf(file_name, "/ucx_shared_mr_uuid_%zu", mm_id >> 1);
-    if (shm_unlink(file_name) != 0) {
-        ucs_warn("unable to unlink the shared memory segment. File name is: %s",
-                 file_name);
-        status = UCS_ERR_SHMEM_SEGMENT;
-        goto err_free_file;
+    /* use the mmid (62 bits uuid) to recreate the file_name for unlink */
+    if (mm_id & UCT_MM_POSIX_SHM_OPEN) {
+        status = uct_posix_set_path(&file_name, 1, NULL, mm_id >> UCT_MM_POSIX_CTRL_BITS);
```

Comment:
i need to send null/path as well. i guess i can send 'path' in anyway and shm_open won't use it.